controllers/controllers.go

```diff
@@ -49,10 +49,24 @@ func (r *instanceReconciler) configureInstance(instance *instances.InstanceInfo,
 	if err != nil {
 		return errors.Wrap(err, "failed to create new nodeconfig")
 	}
-	if err := nc.Configure(); err != nil {
-		return errors.Wrap(err, "failed to configure Windows instance")
+
+	// if instance does not have an associated node, configure it
+	if instance.Node == nil {
+		return nc.Configure()
```

Comment:
This seems to be a change in current behavior. Previously we would have configured the Machine instance irrespective if the node object was present. Are you sure this does not lead to corner cases where we end up with a node object and instance that has not been fully configured?

Reply:
This code is confusing, working on streamlining it.
Right now the function is structured like this:

1) If the instance does not have an associated node, return early and configure it.
2) The instance is confirmed to have a node, if the version annotation has an older value, deconfigure the instance. If it has the current value do nothing and return.
3) All instances that have a node are configured.

Changing it to:
1) If the instance has a node and the version annotation is wrong, deconfigure the instance. If it is correct do nothing and return.
2) Configure all instances

Same result but easier to read